TSynth/TSynth.ino

```diff
@@ -380,514 +294,143 @@ void myNoteOn(byte channel, byte note, byte velocity) {
     //    3         x x x x
     //    4       x       x x
 
-
-    //Voice 1,2,3
-    if (notesOn == 1 || unison == 2) {
-      voice1On(note, velocity, UNISONVOICEMIXERLEVEL);
-      voice2On(note, velocity, UNISONVOICEMIXERLEVEL);
-      voice3On(note, velocity, UNISONVOICEMIXERLEVEL);
-    }
-
-    //Voice 4
-    if (notesOn == 1 || notesOn == 4 || unison == 2)  {
-      voice4On(note, velocity, UNISONVOICEMIXERLEVEL);
-    }
-
-    //Voice 5,6
-    if (notesOn == 1 || notesOn == 3 || unison == 2)  {
-      voice5On(note, velocity, UNISONVOICEMIXERLEVEL);
-      voice6On(note, velocity, UNISONVOICEMIXERLEVEL);
-    }
-
-    //Voice 7
-    if (notesOn == 1 || notesOn == 2 || notesOn == 3 || unison == 2)  {
-      voice7On(note, velocity, UNISONVOICEMIXERLEVEL);
-    }
-
-    //Voice 8
-    if (notesOn == 1 || notesOn == 2 || notesOn == 3 || notesOn == 4 || unison == 2)  {
-      voice8On(note, velocity, UNISONVOICEMIXERLEVEL);
-    }
-
-    //Voice 9
-    if (notesOn == 1 || notesOn == 2 || notesOn == 4 || unison == 2)  {
-      voice9On(note, velocity, UNISONVOICEMIXERLEVEL);
+    if (unison == 2 || notesOn == 1) {
+      for (uint8_t i = 0; i < NO_OF_VOICES; i++) {  
+        voiceOn(i, note, velocity, UNISONNOISEMIXERLEVEL);
+      }
+    } else {
+      // Note: This doesn't take into account notes that have been released.
+      // What happens in the scenario:
+      // 1. note 1 played, voice 0-11 activated
+      // 2. note 2 played, voice 6-11 activated
+      // 3. note 1 released, voice 0-5 disabled
+      // 4. note 3 played, voice 6-11 re-activated
+
+      // TODO: solve by calling getVoiceNo 6 / 4 / 3 times to fetch the best
+      // voices to use based on how many notes are on on. The display code
+      // would need to be updated (or not).

```

Comment:
Note: I think there is a bug in the unison code here.

I also don't understand what "Retrigger" is supposed to do, there doesn't seem to be any special code here for it, just the comment.

Reply:
The Retrigger Voices comment? This shows which voices are retriggered with the new note when playing unison. If three notes are played, voices 4,8 and 9 are retriggered if a fourth note is played.

Reply:
I was mixing this up with the "Retrig" button, thanks for clarifying.